tests/unit/test_collector.py

```diff
@@ -551,44 +551,6 @@ def test_parse_link_handles_deprecated_usage_properly() -> None:
     assert "pkg1-2.0" in parsed_links[1].url
 
 
-def test_parse_links_presents_warning_on_missing_doctype(
-    caplog: pytest.LogCaptureFixture,
-) -> None:
-    html = b'<a href="/pkg1-1.0.tar.gz"></a><a href="/pkg1-2.0.tar.gz"></a>'
-    url = "https://example.com/simple/"
-    page = HTMLPage(html, encoding=None, url=url, cache_link_parsing=False)
-
-    with caplog.at_level(logging.WARN):
```

Comment:
It would be a good idea to keep these tests in and ensure they don't give warnings, with a comment why it matters.

Reply:
So why do we need tests to ensure that there are no warnings, if we are going to remove the code responsible for this?

Reply:
To ensure that N years from now when this whole thing is forgotten that someone will know to not warn when parsing an non-HTML5 document.